### ISSUE TITLE
swarm/start_period Additional property start_period is not allowed

### ISSUE THREAD

**Issue by hartzan** [sameersbn/docker-gitlab#2164]
I think that  version in docker-compose.swarm.yml must be 3.4? If it's 3.3 I got following error:"start_period Additional property start_period is not allowed". I changed it to 3.4 and was able to deploy the stack.

**Comment by solidnerd** [comment]
ping @tiangolo

**Comment by tiangolo** [comment]
Dang it, yes, I added that back from the man `docker-compose.yml` file in the last rebase, I probably missed testing that remotely and only tested it locally :pensive: 

Anyway, the fix is at #2167 .

I just re-deployed a new cluster with those fixes to test everything works :sweat_smile: 

Also, it won't work until the image is re-built with the new changes to generate certificates for the registry.

---

But meanwhile, you can replicate and deploy it following the guide plus these extra steps:

* Clone this repo into the server:

```bash
git clone https://github.com/sameersbn/docker-gitlab.git
```

* Enter into the directory:

```bash
cd docker-gitlab
```

* Create a file `patched.dockerfile`:

```bash
nano patched.dockerfile
```

* Add the contents:

```Dockerfile
FROM sameersbn/gitlab:13.0.0

COPY assets/runtime/ ${GITLAB_RUNTIME_DIR}/
COPY entrypoint.sh /sbin/entrypoint.sh
RUN chmod 755 /sbin/entrypoint.sh
```

* Build it with:

```bash
docker build -t sameersbn/gitlab:13.0.0-patched --file patched.dockerfile .
```

* Then change the `docker-compose.swarm.yml` to use the new patched image:

```bash
sed -i "s|sameersbn/gitlab:13.0.0|sameersbn/gitlab:13.0.0-patched|" docker-compose.swarm.yml
```

* And then you can deploy it normally with:

```bash
docker stack deploy --compose-file docker-compose.swarm.yml gitlab
```

**Comment by hartzan** [comment]
I think we can close this?
